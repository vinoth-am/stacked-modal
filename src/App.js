import React from "react";
import { ModalContex } from "./context/ModalContext";
import { Home } from "./app/screens/Home/Home";

const App = () => {
  return (
    <ModalContex>
      <Home />
    </ModalContex>
  );
};

export default App;
