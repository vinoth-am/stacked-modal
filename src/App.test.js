import { render } from "@testing-library/react";
import App from "./App";
import "@testing-library/jest-dom";

test("App snapshot", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});

test("Validate the child component in App component", () => {
  const { getByText } = render(<App />);
  expect(getByText("Welcome!!")).toBeInTheDocument();
});
