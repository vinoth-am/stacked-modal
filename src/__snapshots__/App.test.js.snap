// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div>
  <div
    class="container"
    data-testid="header"
  >
    <div
      class="header-left"
    >
      <h1>
        Welcome!!
      </h1>
      <h3>
        Use these buttons to open and close the modals.
      </h3>
    </div>
    <div
      class="header-right"
    >
      <button
        class="button primary"
        data-testid="actionButton"
        style="margin: 20px;"
      >
        Open
      </button>
      <button
        class="button secondary"
        data-testid="actionButton"
        style="margin: 20px;"
      >
        Close
      </button>
    </div>
  </div>
  <div
    class="checkbox-container"
  >
    <input
      checked=""
      class="checkbox"
      data-testid="checkbox"
      id="lifo"
      type="checkbox"
    />
    <label
      for="lifo"
    >
      Enable this option to add the margin to the modal for testing the Stacked modal LIFO functinality easily.
    </label>
  </div>
  <div
    data-testid="stackedModal"
  />
</div>
`;
